checkstyle/import-control-share-coordinator.xml

```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE import-control PUBLIC
+        "-//Puppy Crawl//DTD Import Control 1.1//EN"
+        "http://www.puppycrawl.com/dtds/import_control_1_1.dtd">
+<!--
+ Licensed to the Apache Software Foundation (ASF) under one or more
+ contributor license agreements.  See the NOTICE file distributed with
+ this work for additional information regarding copyright ownership.
+ The ASF licenses this file to You under the Apache License, Version 2.0
+ (the "License"); you may not use this file except in compliance with
+ the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+
+<import-control pkg="org.apache.kafka">
+
+    <!-- THINK HARD ABOUT THE LAYERING OF THE PROJECT BEFORE CHANGING THIS FILE -->
+
+    <!-- common library dependencies -->
+    <allow pkg="java" />
+    <!-- no one depends on the server -->
+    <disallow pkg="kafka" />
+
+    <!-- anyone can use public classes -->
+    <subpackage name="coordinator">
+        <subpackage name="share">
+            <allow pkg="org.apache.kafka.common.config" />
+            <allow pkg="org.apache.kafka.coordinator.group" />
+            <allow pkg="org.apache.kafka.common.message" />
+            <allow pkg="org.apache.kafka.common.requests" />
+            <allow pkg="org.apache.kafka.image" />
```

Comment:
This seems to build without the `kafka.image` line. Is it necessary?

Reply:
no it did not:

```
> Task :share-coordinator:checkstyleMain FAILED
[ant:checkstyle] [ERROR] /home/smjn/ws/kafka/share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinatorShard.java:29:1: Disallowed import - org.apache.kafka.image.MetadataDelta. [ImportControl]
[ant:checkstyle] [ERROR] /home/smjn/ws/kafka/share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinatorShard.java:30:1: Disallowed import - org.apache.kafka.image.MetadataImage. [ImportControl]
```